heksher/setting_types.py

```diff
@@ -90,6 +108,21 @@ def __init__(self, name: str):
         """
         super().__init__((int, float))
 
+    def __eq__(self, other):
+        if isinstance(other, IntegerPrimitiveSettingType):
+            return self.types == other.types
+        return NotImplemented
+
+    def __lt__(self, other):
+        if other is setting_type('float'):
+            return True
+        return super().__lt__(other)
+
+    def __gt__(self, other):
+        if other is setting_type('float'):
+            return False
+        return super().__gt__(other)
```

Comment:
```suggestion
    def __lt__(self, other):
        if type(other) == PrimitiveSettingType:
            # since we impose additional requirements over int values, true primitives are lt only if we encompass their type.
            return set(self.types) <= set(other.types)
        return super().__lt__(other)

    def __gt__(self, other):
        if type(other) == PrimitiveSettingType:
           # since we impose additional requirements over int values, we are never greater then true primitives.
            return False
        return super().__gt__(other)
```